pandas/io/excel/_base.py

```diff
@@ -658,6 +658,12 @@ class ExcelWriter(metaclass=abc.ABCMeta):
     datetime_format : str, default None
         Format string for datetime objects written into Excel files.
         (e.g. 'YYYY-MM-DD HH:MM:SS').
+    num_formats : dict, default None
+        Add number formats to columns of type date and datetime.
+        Accepts a dictionary of column name(key) and format strings(value).
+        Formats for columns specified here will take precedence over the
+        ``datetime_format`` and ``date_format`` parameters for those specific columns.
```

Comment:
this shouldn't affect non-number fields at all (e.g. only float, int)

Reply:
I find this extra confusing to try to use this as a date formattter, which we already have a keyword)

Reply:
This change is only for the `date`/`datetime` types. I preferred not to modify in the already existing keywords  `date_format`/`datetime_format` and chose to make a new parameter to make it more generic if the parameter later on needs to be extended to other datatypes as well.

Reply:
I have updated the description with an example